quickwit-cli/src/source.rs

```diff
@@ -275,6 +334,20 @@ fn flatten_json(value: Value) -> Vec<(String, Value)> {
     acc
 }
 
+/// Tries to read a JSON object from a string, assuming the string is an inline JSON object or a
+/// path to a file holding a JSON object.
+async fn sniff_params(params: &str) -> anyhow::Result<Value> {
```

Comment:
Can we use typing here, e.g. a enum of `KafkaParams` or `FileParams`

Reply:
For now, I added a source connectivity check that ensure that the JSON value is a valid `KafkaSourceParams` or `FileParams`. I agree with you at some we need a proper enum `SourceParams`.